hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -135,10 +148,54 @@ public void resume() {
   @Override
   public BackgroundTaskQueue getTasks() {
     BackgroundTaskQueue queue = new BackgroundTaskQueue();
-    queue.add(new DirectoryDeletingService.DirDeletingTask(this));
+    if (taskCount.get() > 0) {
```

Comment:
This means until current tasks are completed, new tasks won't be issued even if there are few idle threads?

Reply:
Looks okay, just wanted to highlight this point.

Reply:
Thanks, it shouldn't make much of a performance difference, as per the default thread config(10) and limit per task(6000), all threads should be able to complete within the deletion interval.